src/transaction/utils.c

```diff
@@ -19,9 +19,18 @@
 #include <stdbool.h>  // bool
 #include <string.h>   // memmove
 
+#if defined(TEST) || defined(FUZZ)
+#define LEDGER_ASSERT(x, y) \
+    if (!(x)) return -1
```

Comment:
Could be done in the unit-test makefile / cmakelist ?
Could be converted to a libc assert?